Scripts/Runtime/Job/AccessControl/IAccessControlledValue.cs

```diff
@@ -0,0 +1,73 @@
+using Unity.Jobs;
+
+namespace Anvil.Unity.DOTS.Jobs
+{
+    /// <summary>
+    /// An interface that describes an access controlled value that can grant read/shared write and exclusive write
+    /// access.
+    /// </summary>
+    /// <typeparam name="T">The type of value being access controlled.</typeparam>
+    public interface IAccessControlledValue<T>
```

Comment:
Yeah I see what you mean. 

What if `IAccessControlledValue` only implemented the `Release` and `ReleaseAsync` methods. 
`IReadAccessControlledValue`, `IExclusiveAccessControlledValue` and `ISharedWriteAccessControlledValue` all inherit from that. 

Then there is final interface which lets you decide what access you want called `IConfigurableAccessControlledValue` that inherits off of `IAccessControlledValue`. 

Naming up for debate, I'm not too invested in that but at least then they all have the common `IAccessControlledValue` as a base.

Reply:
That approach doesn't really get us any free compatibility. It just avoids the duplicate definition of Release/ReleaseAsync.

When I was working on this, trying to avoid duplicating the release definitions I did consider what you've proposed (or using Abstract in the name). The problems are:
 - An object with just Release/ReleaseAsync isn't really a thing. You'd never use that interface other than extending it with another interface
 - Abstract interfaces aren't a thing so there's nothing to prevent using it on its own.

At the end of the day duplicate release definitions aren't a huge deal. If you modify one implementation or one interface all the other ones are going to error so you'll get to the bottom of the issue quickly.

The original comment I was more thinking about a developer that only implements `IAccessControlledValue` (in its current form) and while that type is surely compatible with `IReadAccessControlledValue`'s operations it can't be used unless the authoring developer remembers to also implement that interface.

Maybe this is something we just kick down the road and solve with extension methods, default implementations, or a wrapper in the future.

Reply:
Ah i see, yeah I think it's fine to kick down the road. 

The other problem to add to that list is you can't put a bunch of `I*AccessControlledValue`s in the same collection where * is Read or SharedWrite or ExclusiveWrite etc. But not critical right now.